### ISSUE TITLE
Non-aligned pointer being freed

### ISSUE THREAD

**Issue by deivitaka** [Iterable/swift-sdk#558]
```
IterableCoreDataPersistence.swift line 173
crash_info_entry_0   abort() called
crash_info_entry_1.   myapp(11921,0x16d78f000) malloc: Non-aligned pointer 0x282f6da20 being freed (2)
```

Stack trace
```
Crashed: NSManagedObjectContext 0x28143f8e0
0  libsystem_kernel.dylib         0x7b78 __pthread_kill + 8
1  libsystem_pthread.dylib        0x73bc pthread_kill + 268
2  libsystem_c.dylib              0x2051c abort + 168
3  libsystem_malloc.dylib         0x1ca04 _malloc_put + 550
4  libsystem_malloc.dylib         0x1cc9c malloc_zone_error + 100
5  libsystem_malloc.dylib         0x82f8 szone_free + 464
6  CoreFoundation                 0x451d4 __CFBasicHashDrain + 428
7  CoreData                       0x1ee00 -[NSManagedObjectContext _processRecentChanges:] + 3060
8  CoreData                       0x2da1c -[NSManagedObjectContext save:] + 432
9  myapp                   0x3064c4 <name omitted> + 173 (IterableCoreDataPersistence.swift:173)
10 myapp                   0x306d64 <name omitted> + 4346588516 (<compiler-generated>:4346588516)
11 myapp                   0x306da0 <name omitted> + 80 (CoreDataUtil.swift:80)
12 myapp                   0x306e04 <name omitted> + 4346588676 (<compiler-generated>:4346588676)
13 myapp                   0x306138 <name omitted> + 4346585400 (<compiler-generated>:4346585400)
14 CoreData                       0x1f834 developerSubmittedBlockToNSManagedObjectContextPerform + 156
15 libdispatch.dylib              0x3a2c _dispatch_client_callout + 20
16 libdispatch.dylib              0x12c90 _dispatch_lane_barrier_sync_invoke_and_complete + 56
17 CoreData                       0x1fa94 -[NSManagedObjectContext performBlockAndWait:] + 264
18 myapp                   0x305f38 performAndWait + 79 (CoreDataUtil.swift:79)
19 myapp                   0x306adc schedule + 23 (IterableTaskScheduler.swift:23)
20 myapp                   0x316dc0 sendIterableRequest + 270 (<compiler-generated>:270)
21 myapp                   0x314c34 track + 4346645556 (<compiler-generated>:4346645556)
22 myapp                   0x32a6e0 <name omitted> + 143 (RequestHandler.swift:143)
23 myapp                   0x2f8e34 <name omitted> + 4346531380 (<compiler-generated>:4346531380)
24 myapp                   0x2cc690 <name omitted> + 105 (Pending.swift:105)
25 myapp                   0x2cc4a8 <name omitted> + 4346348712 (<compiler-generated>:4346348712)
26 myapp                   0x2ca594 onSuccess + 4346340756 (<compiler-generated>:4346340756)
27 myapp                   0x2c0d78 flatMap + 25 (Pending.swift:25)
28 myapp                   0x2f8960 inBackgroundThread + 175 (Pending.swift:175)
29 myapp                   0x335664 track + 4346779236 (RequestHandler.swift:4346779236)
```

**Comment by tapashmajumder** [comment]
@deivitaka 
I was not able to repro this but I think I know what is causing this.
Can you please try the `tapash/mob-4488-core-data` branch and see if the issue is resolved?

**Comment by tapashmajumder** [comment]
Hi. This should be fixed with release 6.4.4. Please open a ticket if this is not fixed.
